src/main/java/com/getcloudcherry/survey/SurveyActivity.java

```diff
@@ -257,63 +246,293 @@ void filterQuestions(SurveyResponse iResponse) {
-                @Override
-                public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
-                    try {
-                        aProgressDialog.dismiss();
-                        finishSurvey(aAlert);
-                    } catch (Exception e) {
-                        e.printStackTrace();
+    /**
+     * API call to submit answers
+     */
+    public void submitAnswers() {
+        final ProgressDialog aProgressDialog = new ProgressDialog(SurveyActivity.this);
+        aProgressDialog.setTitle("Submitting survey");
+        aProgressDialog.setMessage("Please wait...");
+        aProgressDialog.setCancelable(false);
+        aProgressDialog.show();
+        Call<ResponseBody> aCall = SurveyClient.get().postAnswerAll(SurveyCC.getInstance().getSurveyToken(), RecordAnswer.getInstance().getAnswers());
+        aCall.enqueue(new Callback<ResponseBody>() {
+            @Override
+            public void onResponse(Call<ResponseBody> call, Response<ResponseBody> response) {
+                try {
+                    aProgressDialog.dismiss();
+                    finishSurvey();
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            }
+
+            @Override
+            public void onFailure(Call<ResponseBody> call, Throwable t) {
+                try {
+                    aProgressDialog.dismiss();
+                    if (aAlertDialog != null) {
+                        aAlertDialog.cancel();
                     }
+                    showAlertRetryCallback(RETRY_SUBMIT, getString(R.string.alert_message_survey_submit_failed), SurveyActivity.this);
+                } catch (Exception e) {
+                    e.printStackTrace();
                 }
+            }
+        });
+    }
 
-                @Override
-                public void onFailure(Call<ResponseBody> call, Throwable t) {
-                    try {
-                        aProgressDialog.dismiss();
-                        if (aAlert != null) {
-                            aAlert.cancel();
-                        }
-                        showAlertRetryCallback(getString(R.string.alert_message_survey_submit_failed), SurveyActivity.this, new DialogInterface.OnClickListener() {
-                            @Override
-                            public void onClick(DialogInterface dialogInterface, int i) {
-                                submitAnswers(null);
-                            }
-                        });
-                    } catch (Exception e) {
-                        e.printStackTrace();
+    /**
+     * API call to submit partial answers
+     *
+     * @param isLastPage boolean to check if view pager is on last page
+     * @param iAnswer    array list of answer object as per API documentation
+     */
+    public void submitAnswerPartial(final boolean isLastPage, ArrayList<Answer> iAnswer) {
```

Comment:
In First Partial API Request, prefill answers should also be appended along with first question's Answer, else it won't be saved anywhere.